extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationSearch.js

```diff
@@ -0,0 +1,36 @@
+$(function ($) {
+	'use strict';
+
+	var $globalNav = $('.wds-global-navigation'),
+		$searchInput = $globalNav.find('.wds-global-navigation__search-input'),
+		$searchSubmit = $globalNav.find('.wds-global-navigation__search-submit'),
+		placeHolderText = $searchInput.attr('placeholder'),
+		activeSearchClass = 'wds-search-is-active';
+
+	$searchInput.on('focus', function () {
+		if (!$globalNav.hasClass(activeSearchClass)) {
+			$globalNav.addClass(activeSearchClass);
+			$searchInput.attr('placeholder', $searchInput.data('active-text'));
+		}
+	});
+
+	// Make sure default state is correct when navigating back in Chrome
+	if ($searchInput.val().length > 0 && $searchSubmit.prop('disabled')) {
+		$searchSubmit.prop('disabled', false);
+	}
+
+	$searchInput.on('input', function () {
+		var textLength = $(this).val().length;
+
+		if (textLength > 0 && $searchSubmit.prop('disabled')) {
+			$searchSubmit.prop('disabled', false);
+		} else if (textLength === 0) {
```

Comment:
Yeah, I meant that. Although I'm not sure what am I missing. When user removes the text, `textLength > 0` will be evaluated to false, so button will get re-disabled.

BTW: I'm have completely no idea which events are fired when user clicks `button type=reset`. can you make sure if it fires `input` event so we won't end up with enabled button when user close search?